### DISCUSSION TITLE
Best way to tag files of a project together?

### DISCUSSION THREAD

**Discussion by Sundar9987** [community/community#22116] [New to GitHub]
<p>Team - I am trying to integrate github with Jenkins. The question I have is - how do i pull only a set of files from github repository? I dont want all the files in the repository for the CI - build phase. For example, my repository would have files for Project X,Y &amp; Z, but I need only the files that belong to project Z.</p>
<ol>
<li>How do i tag the files associated with project Z in the github repository (master branch)?</li>
<li>How to get tell the jenkins to only pick those files from the github repo?</li>
</ol>
<p>Thanks,<br>
Sundar</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>It sounds like you want to do a sparse checkout. You can find basic information on how to it in the topic on how to download a single folder from a repo:</p>
<aside class="quote quote-modified" data-post="2" data-topic="278">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/how-can-i-download-a-specific-folder-from-a-github-repo/278/2">How can I download a specific folder from a GitHub repo?</a> <a class="badge-wrapper  bullet" href="/c/github-help/how-to-use-git-and-github/21"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #ffd1ac;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Ask and answer questions regarding Git and GitHub.">How to use Git and GitHub</span></a>
  </div>
  <blockquote>
    If you are willing to use git to do this, you can do a sparse checkout. 
We are going to start by creating an empty git repo locally, and then add the repository we want as a remote. This allows us to easily just checkout the folder we want 
mkdir directoryName
cd directoryName
git init
git remote add origin -f repoUrl

After that, we want to add the directory to  
.git/info/sparse-checkout

(this is essentially the opposite of the gitignore file). 
Then we can  
git pull origin master

and it s…
  </blockquote>
</aside>

<p>If you want something more complex than a single folder, you can find out more information <a href="https://git-scm.com/docs/git-sparse-checkout">in the documentation on the <code>git sparse-checkout command</code></a>.</p>
<p>I hope that helps!</p>
